### PR TITLE
Move `SnippetViewSet.list_filter` and `filterset_class` definition to `ModelViewSet`

### PR DESCRIPTION
**Pull request by laymonage**
Part of #10740.

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [x] For new features: Has the documentation been updated accordingly?

As with #10796, I'd suggest spinning up the testapp project and play around with the "Feature complete toys" model.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: wagtail/test/settings.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from django.contrib.messages import constants as message_constants
 from django.utils.translation import gettext_lazy as _
 
-DEBUG = False
+DEBUG = True
```
Comment (gasman) [inline wagtail/test/settings.py:6]: Why is this change needed?

**Review by gasman — CHANGES_REQUESTED** [review]
Just one quibble, otherwise looks good!

File: wagtail/test/settings.py (hunk 1 above)
Comment (laymonage) [reply wagtail/test/settings.py:6]: Oops, it's not, just leftover from testing 😆

Though honestly I'd really like us to change this (in a separate PR). This was added in #3381, but I'm not convinced it gave any speed up at all. From https://docs.djangoproject.com/en/stable/topics/testing/overview/#other-test-conditions:

> Regardless of the value of the [`DEBUG`](https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-DEBUG) setting in your configuration file, all Django tests run with [`DEBUG`](https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-DEBUG)=False. This is to ensure that the observed output of your code matches what will be seen in a production setting.

Changing this to `True` would allow us to easily run the testapp locally (without having to configure static files) for testing PRs, since we can reuse the code (e.g. things like the ViewSet definitions) instead of having to modify bakerydemo.
